scripts/build_osm_network.py

```diff
@@ -830,8 +830,12 @@ def built_network(inputs, outputs, geo_crs, distance_crs):
     country_list = input
     bus_country_list = buses["country"].unique().tolist()
 
-    if len(bus_country_list) != len(country_list):
-        no_data_countries = set(country_list).difference(set(bus_country_list))
+    # it may happen that bus_country_list contains entries not relevant as a country name (e.g. "not found")
+    # difference can't give negative values; the following will return only releant country names
+    no_data_countries = set(country_list).difference(set(bus_country_list))
```

Comment:
may be worth using symmetric_difference here

Reply:
Because we can't handle two difference cases as symmetrical:

1) `set(country_list).difference(set(bus_country_list))` = countries from the `countries` parameter of the config which don't have any data to restore a buses dataframe [meaning that we need to generate some data for such areas]

2) `set(bus_country_list).difference(set(country_list)`) = countries from the buses dataframe which are not in the `countries` from the config and hence were not requested by the user to be included into the model [which basically means that something went wrong in the workflow before]

Initially the following code chunk was intended to address the first situation but the condition captured both of them being `len(bus_country_list) != len(country_list)` ( see https://github.com/pypsa-meets-earth/pypsa-earth/blob/fa799bf0723496a3ac5ec50feb9a2ffe2876d758/scripts/build_osm_network.py#L833 ). That lead to troubles when a non-standard code appeared in the buses list. So, I had to introduce a fix exactly to avoid mixing two situations. (Sorry for not being clear enough when explaining it in our discussion!)

Reply:
My proposal would be to use:
`difference_countries = set(country_list).symmetric_difference(set(bus_country_list))`
difference_countries should be empty if the two sets match, othewise difference_countries contains the items that are missing in one set or in the other one

if country list is ["AG"]and bus_country_list is ["AG", "Something"], the current revision doesn't catch the difference while the previousone using len does

Reply:
Agree, if the case you mention arise, it should be fixed

But the approach should be different. Would it be probably a good idea to have an additional check on `set(bus_country_list).difference(set(country_list))` and throw an error if it happens? Because this would mean that our attempt to fix it in `build_shapes` has failed